src/Analyzer.TemplateProcessor.UnitTests/ArmTemplateProcessorTests.cs

```diff
@@ -240,6 +240,115 @@ public void ParseAndValidateTemplate_ValidTemplateWithExpressionInResourceProper
+                            ""frontendIPConfiguration"": {
+                                ""id"": ""Microsoft.Network/loadBalancers/loadBalancer/frontendIPConfigurations/config""
+                            },
+                            ""frontendPort"": ""[copyIndex(2200)]"",
+                            ""protocol"": ""Tcp""
+                        },
+                        ""type"": ""Microsoft.Network/loadBalancers/inboundNatRules""
+                    }
+                ],
+                ""outputs"": {}
+            }";
+
+            ArmTemplateProcessor armTemplateProcessor = new ArmTemplateProcessor(templateJson);
+
+            var parameters = new InsensitiveDictionary<JToken>();
+
+            var metadata = new InsensitiveDictionary<JToken>
+            {
+                { "subscription", new JObject(
+                    new JProperty("id", "/subscriptions/00000000-0000-0000-0000-000000000000"),
+                    new JProperty("subscriptionId", "00000000-0000-0000-0000-000000000000"),
+                    new JProperty("tenantId", "00000000-0000-0000-0000-000000000000")) },
+                { "resourceGroup", new JObject(
+                    new JProperty("id", "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroupName"),
+                    new JProperty("location", "westus2"),
+                    new JProperty("name", "resource-group")) },
+                { "tenantId", "00000000-0000-0000-0000-000000000000" }
+            };
+
+            Dictionary<string, string> expectedMapping = new Dictionary<string, string> {
+                { "resources[0]", "resources[1]" },
+                { "resources[1]", "resources[0]" },
+                { "resources[2]", "resources[0]" },
+                { "resources[3]", "resources[2]" },
+                { "resources[4]", "resources[2]" },
+            };
+
+            Template template = armTemplateProcessor.ParseAndValidateTemplate(parameters, metadata);
+
+            Assert.AreEqual(expectedMapping.Count, template.Resources.Length);
+
+            // First resource should be the resource without a copy index specified
+            Assert.AreEqual("resourceWith/NoCopyLoop", template.Resources.First().Name.Value);
+
+            Assert.AreEqual("name/SSH-VMNamePrefix-1", template.Resources[2].Name.Value);
+            Assert.AreEqual("[concat('name', '/', 'SSH-', 'VMNamePrefix-', copyIndex())]", template.Resources[2].OriginalName);
+            Assert.AreEqual(2201, template.Resources[2].Properties.Value.InsensitiveToken("frontendPort").Value<int>());
+
+            Assert.AreEqual("name/CopyLoop2-SSH-VMNamePrefix-1", template.Resources.Last().Name.Value);
+            Assert.AreEqual("[concat('name', '/', 'CopyLoop2-SSH-', 'VMNamePrefix-', copyIndex())]", template.Resources.Last().OriginalName);
+            Assert.AreEqual(2201, template.Resources.Last().Properties.Value.InsensitiveToken("frontendPort").Value<int>());
```

Comment:
I don't think these assertions provide much value.  Since we have the path mapping between expanded and original templates (and that's all that's used), the order of the resources in the expanded template shouldn't matter in the end.

Reply:
I can remove them and just leave the dictionary comparison.